pinot-segment-local/src/main/java/org/apache/pinot/segment/local/io/writer/impl/MutableOffHeapByteArrayStore.java

```diff
@@ -218,10 +223,10 @@ public int add(byte[] value) {
     int index = buffer.add(value);
     if (index < 0) {
       // Need to expand the buffer
-      int currentBufferSize = buffer.getSize();
-      if ((currentBufferSize << 1) >= 0) {
+      long nextBufferSize = buffer.getSize() << 1;
```

Comment:
Do you need to cast before shifting?

Reply:
Just realize we don't need to change this. Checking if it `>= 0` after shifting is always correct